### DISCUSSION TITLE
Inverted color variables: are these correct?

### DISCUSSION THREAD

**Discussion by hugopeek** [fomantic/Fomantic-UI#2086] [Elements (button, image, label, loader, ...)] [answered]
I was just comparing my custom variables with FUI core variables and noticed the following additions:

https://github.com/fomantic/Fomantic-UI/blob/f451d1549c1ec53f99c9d64f8482f913311e1a36/src/themes/default/globals/site.variables#L302-L303

I'm a little confused by them being inverted... Logically, I would expect these variables to be primaryColor and secondaryColor, so they match the rest of the list (all dark colors). Setting them to primary and secondary also doesn't seem to change anything for me. I found references in label and button components, but they still look the same, whatever I enter there. The colors.less file also uses them but again, I can't figure out how or where.

So not really an issue, just wondering what they are doing.

For reference, these variables were introduced here:

https://github.com/fomantic/Fomantic-UI/commit/7338ad2a295c5460bdd97f261c36ee85032d5bd3#diff-7124c2e79f2279a834c8ce10fcfb9c7727c91770bc49cae2427f22b6609f4585

**Comment by lubber-de** [comment, marked as answer]
`primaryColor` and `secondaryColor` are (unfortunately) often used for background colors and are dark (black and blue), 

https://github.com/fomantic/Fomantic-UI/blob/f451d1549c1ec53f99c9d64f8482f913311e1a36/src/themes/default/globals/site.variables#L56-L60

so the `primaryTextColor` and `secondaryTextColor` (used inside buttons/labels) cannot be the same. The desired color is white, that's why we have choosen to use `invertedTextColor` here

They are defined as `text` colors, which is used inside buttons/labels in the central `colors.less` file here

https://github.com/fomantic/Fomantic-UI/blob/f451d1549c1ec53f99c9d64f8482f913311e1a36/src/themes/default/globals/colors.less#L14

I have to admit, this is somewhat inconsistent and would need a cleanup over all components...
